### PR TITLE
feat(swagger): Support x-doc-exclude and x-doc-host

### PR DESCRIPTION
**Pull request by bhaskarmelkani**
Updates in the PR.
### 1. Add support for parsing swagger file before passing it to `swager-ui` or `swagger-to-html`.
  **Why?**
 There are use cases when you want to process swagger file before using it. Some use cases are removing some paths/operations/definitions that you don't want to expose, updating the host, etc.

Now you can decorate the swagger object before using it.

Some provided decorators : 
#### `x-doc.exclude`
This is decorator is used when you want to remove some definitions/objects from swagger.
You can exclude following definitions.
  * operations: GET, POST, etc. operations for a specific path
  * parameters: Remove parameters that you don't want to expose.
  * paths: Remove complete path from the spec.
  * security: Remove security definitions that are not required for someone who is reading the documentation.

Eg - Some swagger file.
```yaml
swagger: "2.0"
.
.
paths:
  /pets:
    get:
      xdoc:
        exclude: true
.
.
```
The `get` operation from `/pets` will be removed, after the decorator is applied. 

#### `x-doc: host`
```yaml
xdoc:
  host: example.com
```
`xdoc.host` will overwrite the host.

You can add custom decorators in `lib/nodejs/swagger/decorators`

### 2. Support remote links for swagger spec in `swagger-ui` and `swagger-to-html`.
You can pass http links to both the tags.

### 3. Remove `-x` from examples for request in `swagger-to-html`.

### CODE REVIEW

**Comment by bhaskarmelkani** [comment]
👍

File: plugins/swagger-routes.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const util = require('../lib/nodejs/hexo-util');
+
+const getFilter = (hexo) => {
+
+  const swaggerStore = require('../lib/nodejs/swagger-store')({hexo});
+
+  return () => {
+    const routes = swaggerStore.getRoute();
```
Comment (rbarilani) [inline plugins/swagger-routes.js]: @bhaskarmelkani let's call this method getRoutes

Comment (bhaskarmelkani) [reply plugins/swagger-routes.js]: Okay

File: plugins/swagger-routes.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        }else{
+          hexo.route.remove(route);
+        }
+      })
+  }
+}
+
+
+module.exports = ({hexo}) => {
+  const {themeConfig} = util({hexo});
```
Comment (rbarilani) [inline plugins/swagger-routes.js]: @bhaskarmelkani why do you need this?

Comment (bhaskarmelkani) [reply plugins/swagger-routes.js]: Removed. First I was adding some configs for routes then removed it for simplicity.

**Comment by rbarilani** [comment]
👍

**Pull request merged**